### PR TITLE
crypto: audit SuiAddress and ObjectID

### PR DESCRIPTION
**Pull request by joyqvq**
## Description 

Audit all methods for SuiAddress and ObjectID in base_types.rs: added docs, removed unused, clean up test only methods behind feature flag. 
## Test Plan 

Existing tests. 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-types/src/unit_tests/base_types_tests.rs [hunk 1]
```diff
@@ -124,14 +124,8 @@ fn test_object_id_conversions() {}
...
 fn test_object_id_display() {
     let hex = SAMPLE_ADDRESS;
-    let upper_hex = SAMPLE_ADDRESS.to_uppercase();
-
     let id = ObjectID::from_str(hex).unwrap();
     assert_eq!(format!("{:?}", id), format!("0x{hex}"));
-    assert_eq!(format!("{:X}", id), upper_hex);
-    assert_eq!(format!("{:x}", id), hex);
-    assert_eq!(format!("{:#x}", id), format!("0x{hex}"));
-    assert_eq!(format!("{:#X}", id), format!("0x{upper_hex}"));
```
Comment (joyqvq) [inline crates/sui-types/src/unit_tests/base_types_tests.rs:134]: i dont think we should/need to support all these formatting but to prefer one 0xhex

**Review by benr-ml — APPROVED** [review]

**Review by kchalkias — APPROVED** [review]
Thanks @joyqvq

File: crates/sui-types/src/base_types.rs [hunk 2]
```diff
@@ -475,10 +507,12 @@ impl From<&PublicKey> for SuiAddress {
     }
 }
 
-/// A MultiSig address is the first 20 bytes of the hash of
-/// `flag_MultiSig || threshold || flag_1 || pk_1 || weight_1 || ... || flag_n || pk_n || weight_n`
-/// of all participating public keys and its weight.
 impl From<MultiSigPublicKey> for SuiAddress {
+    /// Derive a SuiAddress from [struc MultiSigPublicKey]. A MultiSig address
```
Comment (kchalkias) [inline crates/sui-types/src/base_types.rs:511]: from struc**t**

**Review by kchalkias — COMMENTED** [review]
Indexer complains for a From derivation, it seems related to this PR.

**Pull request merged**